src/orqviz/fourier.py

```diff
@@ -0,0 +1,243 @@
+                the coefficient for 0, then the coefficients of positive frequencies in
+                increasing order by frequency, then the coefficients of negative
+                frequences in increasing order.
+            end_points_x: End points of the scan along the x-direction.
+            end_points_y: End points of the scan along the y-direction.
+    """
+    if n_steps_y is None:
+        n_steps_y = n_steps_x
+    if end_points_y is None:
+        end_points_y = end_points_x
+    scan2D_result = perform_2D_scan(
+        origin,
+        loss_function,
+        direction_x=direction_x,
+        direction_y=direction_y,
+        n_steps_x=n_steps_x,
+        n_steps_y=n_steps_y,
+        end_points_x=end_points_x,
+        end_points_y=end_points_y,
+    )
+    fourier_result = np.fft.rfft2(scan2D_result.values, norm="forward")
+    return FourierResult(fourier_result, end_points_x, end_points_y)
+
+
+def plot_2D_fourier_result(
+    result: FourierResult,
+    max_freq_x: float = None,
+    max_freq_y: float = None,
+    show_negative_frequencies: bool = False,
+    fig: Optional[matplotlib.figure.Figure] = None,
+    ax: Optional[matplotlib.axes.Axes] = None,
+    **plot_kwargs,
+):
+    """Plots a 2D fourier result.
+    Args:
+        result: Fourier result to be plotted.
+        max_freq_x: Maximum frequency to be plotted along the x-direction.
+            If None, all the output frequencies in the result are plotted.
+        max_freq_y: Maximum frequency to be plotted along the y-direction.
+            If None, all the output frequencies in the result are plotted.
+        show_negative_frequencies: only plot positive frequencies if False
+        fig: Matplotlib figure to perfom a plot on. If None, a new figure
+            and axis are created from the current figure. Defaults to None.
+        ax: Matplotlib axis to perform plot on. If None, a new axis is created.
+            Defaults to None.
+        plot_kwargs: kwargs for plotting with matplotlib.pyplot.pcolormesh
+            (plt.pcolormesh)
+
+    Note: Fourier coefficients are complex numbers. However, due to the inability to
+    plot complex numbers on real axes, the magnitudes of the coefficients are plotted.
+    This means that phase has no influence on the visual output of the plot.
+    """
+    plottable_result = np.abs(result.values)
+    Nx = result.values.shape[1]
+    Ny = result.values.shape[0]
+    if max_freq_y is None:
+        max_freq_y = min(Ny // 2, max_freq_x or np.inf)
+    if max_freq_x is None:
+        max_freq_x = Nx - 1
+
+    # normalize frequencies for range bigger than 1 period of 2pi
+    # (note that this is different from normalizing the coefficient magnitudes)
+    norm_x = (result.end_points_x[1] - result.end_points_x[0]) / (2 * np.pi)
+    norm_y = (result.end_points_y[1] - result.end_points_y[0]) / (2 * np.pi)
+    if max_freq_x > (Nx - 1) / norm_x:
+        warnings.warn(
```

Comment:
There's a lot of cases here in if statements which are not covered by tests, could please add some tests for these?